dwh/query_lib.py

```diff
@@ -262,15 +271,25 @@ def _flatten_obs(obs: DataFrame, code_system: str = None) -> DataFrame:
       - `dateTime` from the input's `effective.dateTime`
     """
     sys_str = '="{}"'.format(code_system) if code_system else 'IS NULL'
+    merge_udf = F.UserDefinedFunction(
+        lambda d, v: merge_date_and_value(d, v), T.StringType())
```

Comment:
@bashir2  any reasons why we are merging value/code + date ? `5090AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA_min_date='2019-03-30T08:37:06+00:00'` Whouldn't it be idea

Reply:
The idea here is to find what is the Value/ValueCode corresponding to the first (or last) observation. What we are doing here is add the date of the observation as a prefix for Value/ValueCode and then in aggregations on line 310-313 below find that value (corresponding to first/last observation). You can see examples in the 3rd cell [here](https://github.com/GoogleCloudPlatform/openmrs-fhir-analytics/blob/master/dwh/test_query_lib.ipynb).